framework/vst/plugininstance.cpp

```diff
@@ -0,0 +1,286 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "plugininstance.h"
+
+#include "plugin.h"
+#include "log.h"
+
+using namespace mu::vst;
+using namespace Steinberg;
+using namespace Steinberg::Vst;
+
+PluginInstance::PluginInstance(const Plugin* plugin)
+    : m_valid(false), m_active(false), m_parameters(), m_host(),
+    m_effectClass(plugin->m_effectClass),
+    m_factory(plugin->m_factory),
+    m_controller(nullptr),
+    m_componentCP(nullptr), m_controllerCP(nullptr),
+    m_audioProcessor(nullptr)
+{
+    init();
+    connect();
+    initAudioProcessor();
+    m_valid = m_component && m_controller && m_audioProcessor;
+
+    initParameters();
+}
+
+PluginInstance::~PluginInstance()
+{
+    if (m_controller) {
+        m_controller->terminate();
+    }
+
+    if (m_component) {
+        m_component->terminate();
+    }
+}
+
+bool PluginInstance::createView()
+{
+    IF_ASSERT_FAILED(isValid()) {
+        LOGE() << "plugin instance is not valid";
+        return false;
+    }
+
+    auto view = m_controller->createView(ViewType::kEditor);
+    if (!view) {
+        LOGE() << "plugin hasn't view";
+        return false;
+    }
```

Comment:
Better use Ret type for returning code error and message

Reply:
`bool PluginInstance::createView()` has return type bool temporarily. It should return something like PluginViewInterface.